middleware.ts

```diff
@@ -81,8 +81,7 @@ export async function middleware(request: NextRequest) {
     // store redirection status in custom header as referrer seems missing from headers in RSC
     // isRedirected = params is present + previous page is coming from site
     const referer = requestHeaders.get('referer') || '';
-    const baseURL =
-      process.env.NEXT_PUBLIC_BASE_URL || 'https://annuaire-entreprises';
+    const baseURL = process.env.NEXT_PUBLIC_BASE_URL || 'http://localhost:3000';
```

Comment:
Sounds bug prone, default should be prod url rather than local. If one forgot to declare `NEXT_PUBLIC_BASE_URL` it is OK if local fails, but not if prod fails because of it. I prefer 

`const baseURL = process.env.NEXT_PUBLIC_BASE_URL || 'https://annuaire-entreprises.data.gouv.fr';`

Reply:
Actually not sure`prodBaseUrl` is a good idea

Reply:
I did it this way because most of the time in other repos it is written like that.

Reply:
I renamed to getAbsoluteSiteUrl, good idea !

Reply:
I'm okay with the two versions

Reply:
I see but the way we use it, it means that if we forget it while provisionning a new machine, it will break prod. 

I would consider it a good practise to use loacalhost if for instance we would  use it to log errors or stats. Better log them as localhost by default in order not to polluate our analytics

Reply:
I prefer to have a working default otherwise we have to add tests to our provisionning scripts